vendor/bat-native-ledger/src/bat_client.cc

```diff
@@ -506,32 +531,46 @@ void BatClient::setGrantCallback(bool success,
   ledger_->OnGrantFinish(ledger::Result::LEDGER_OK, grant);
 }
 
-void BatClient::getGrantCaptcha() {
+void BatClient::getGrantCaptcha(bool isAndroid) {
   std::vector<std::string> headers;
   headers.push_back("brave-product:brave-core");
-  auto request_id = ledger_->LoadURL(braveledger_bat_helper::buildURL((std::string)GET_PROMOTION_CAPTCHA + ledger_->GetPaymentId(), PREFIX_V2),
-   headers, "", "",
+
+  auto request_id = ledger_->LoadURL(braveledger_bat_helper::buildURL((std::string)(isAndroid ? GET_PROMOTION_ATTESTATION : GET_PROMOTION_CAPTCHA) +
+      ledger_->GetPaymentId(), isAndroid ? PREFIX_V1 : PREFIX_V2),
+      headers, "", "",
       ledger::URL_METHOD::GET, &handler_);
   handler_.AddRequestHandler(std::move(request_id),
                              std::bind(&BatClient::getGrantCaptchaCallback,
                                        this,
                                        _1,
                                        _2,
-                                       _3));
+                                       _3,
+                                       isAndroid));
 }
 
 void BatClient::getGrantCaptchaCallback(bool success,
                                         const std::string& response,
-                                        const std::map<std::string, std::string>& headers) {
+                                        const std::map<std::string, std::string>& headers,
+                                        bool isAndroid) {
```

Comment:
same thing here. I'm not understanding why the ledger logic needs to be different for android. The safetynet stuff is part of `chrome/browser/android` and isn't necessarily relevant to someone else using ledger on android in a different context

Reply:
see above for specifying what we want to do vs the platform we are doing it on within ledger. Remember that the ledger code is still meant to be an independent library to be used by others